rand_core/src/impls.rs

```diff
@@ -207,6 +207,30 @@ impl<R: BlockRngCore + fmt::Debug> fmt::Debug for BlockRng<R> {
     }
 }
 
+impl<R: BlockRngCore> BlockRng<R> {
+    /// Create a new `BlockRng` from an existing RNG implementing
+    /// `BlockRngCore`. Results will be generated on first use.
+    pub fn new(core: R) -> BlockRng<R>{
+        let results_empty = R::Results::default();
+        BlockRng {
+            core,
+            index: results_empty.as_ref().len(),
+            results: results_empty,
+        }
+    }
+
+    /// Return a mutable reference the wrapped `BlockRngCore`.
+    pub fn inner(&mut self) -> &mut R {
+        &mut self.core
```

Comment:
Really we should have mutable *and* immutable accessors, i.e. `fn inner(&self) -> &R` and `fn inner_mut`.

I'm surprised you didn't call this `core` though (you can shadow the name).

Reply:
I did initially, but I then thought this would make more sense for users, who don't see the field name.

Reply:
I just thought it aligned well with `BlockRngCore` but doesn't matter.